kiosk/kiosk_browser/browser_widget.py

```diff
@@ -77,6 +80,18 @@ def reload(self):
         if self._reload_timer.isActive():
             self._reload_timer.stop()
 
+    def hard_refresh(self):
+        """ Clear cache, then reload.
+
+        Does not affect cookies or localstorage contents.
+
+        NOTE This clears the entire HTTP cache, assumed to be OK as the kiosk targets a specific page.
+        """
+        logging.info(f"Clearing HTTP cache (hard refresh)")
+        QtWebEngineWidgets.QWebEngineProfile.defaultProfile().clearHttpCache()
```

Comment:
Thanks, that sounds like it might be better!

One downside is that if someone presses Ctrl-Shift-R when the controller interface is open, they would not clear the cache of the right page. (Right?)

Reply:
e68cb1c7ee4ab5b0afcc6738a44a22a2d38c9c1e